### PR TITLE
Storyq-28 Text Display Bugs

### PR DESCRIPTION
**Pull request by tealefristoe**
Jira stories:
https://concord-consortium.atlassian.net/browse/STORYQ-28
https://concord-consortium.atlassian.net/browse/STORYQ-21
https://concord-consortium.atlassian.net/browse/STORYQ-22

This PR addresses several bugs related to text being displayed and highlighted properly at various stages of the StoryQ process. Text should now be displayed and highlighted properly in almost all cases:
- With regular as well as token (single word) features.
- At all stages in the process (after creating features, after training a model, and after testing a model).
- When cases are selected in the feature or text tables.

I ran out of time before I could fix text display when multiple models have been trained and the user is in the test panel. And there are probably other edge cases that still don't work properly.

There are a couple of general bugs that were causing text to display incorrectly before.
- Codap v3 returns strings for all case values, even booleans. To accommodate for this, there are a couple of places where we're explicitly looking for the string "true" in StoryQ now. Ultimately, it would be better for Codap v3 to return different types of values, but I wasn't ready to make that change, which could cause problems for other plugins. When Codap is fixed, it will probably break StoryQ here. I left comments in the code to hopefully make fixing that problem easier.
- Features are stored in a table in Codap, but each feature has at least two cases in two different collections, and possibly more. You'd think that the parent case would be used as the feature's actual case, but that isn't always (ever?) how it's represented in StoryQ. I'm not sure if this was an inconsistency in the original plugin, or whether a single error was introduced at some point that then permeated throughout the codebase. But in any case, there were several places where I changed the feature's case id to be for a (usually the) child case, rather than the parent. I think it would be much better to consistently use the parent case, but I'm not sure how much work it would take to make this switch, or if there's actually a good reason to use the child case.
- There are actually two kinds of features: regular features and "tokens" (created by single word features). There's a lot of ambiguity between these, with both features and tokens sometimes called features or even tokens in the codebase. For users, they seem like they're all features. They have to be handled differently in the code, and there were times where the code just didn't handle tokens at all. I tried to improve this situation, being more explicit about features vs tokens and handling both of them whenever it was needed, but there are still probably places where it could be improved and it will almost certainly cause confusion for future engineers.

Additional minor improvements:
- Train and step buttons in the training panel now appear disabled when they are disabled.
- Colors in text section titles are improved. This does not match the design spec, but it's an improvement over what they were previously (like neon green, very difficult to read) and I doubt I'll have time to actually address this fully.
- As usual, I made layouts more consistent and simplified code whenever possible.

### CODE REVIEW

File: src/managers/testing_manager.ts [hunk 1]
```diff
@@ -129,7 +127,8 @@ export class TestingManager {
 				// The column features are names of attributes we expect to find having values true or false
 				tTokens.forEach((iToken, iIndex) => {
 					if (iToken.formula !== '') {
-						if (iCase.values[iToken.name]) {
+						// Codap v3 currently returns strings, even for booleans, so we have to compare the string for now
+						if (iCase.values[iToken.name] === "true") {
```
Comment (scytacki) [inline src/managers/testing_manager.ts:131]: If you wanted to be forwards compatible, you could change this to:
```suggestion
						if (iCase.values[iToken.name] === "true" || iCase.values[iToken.name] === true) {
```

**Review by scytacki — APPROVED** [review]
Looks good to me.

Just one suggestion about making it forward compatible by handling `true`.

Also I've noticed that the spacing seems inconsistent at least when reviewing the changes in the GitHub web interface. I'm guessing the files are using tabs in some cases, perhaps 2 tabs for one indent. So the indentation is 8 characters in the GitHub web ui. This makes it harder to review. No urgency, but I'd recommend switching to a 2 space indentation (no tabs).

**Comment by tealefristoe** [comment]
@scytacki I think I will wait on fixing the tabs/spaces for now. Would setting an eslint rule fix the issue?

**Comment by scytacki** [comment]
@tealefristoe yes I'd guess there is a ts or js stylistic plugin that has a rule for it. However I don't see it explicitly configured in the projects I've checked. Perhaps we don't have it configured because it needs to have lots of options inorder to automatically indent things "correctly". 

The most generic way to configure for new code is with an `.editorconfig`. We have one in CLUE: https://github.com/concord-consortium/collaborative-learning/blob/master/.editorconfig
You can read more about them here: https://editorconfig.org/
You have to have a plugin in VSCode for it to be picked up: https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig
Github will also read this file and at least use it for the tab width setting (which we don't actually need if we use spaces)

I don't know if that is the best way for us to setup things, but it seems good to me.

I also don't know the best way to go through and fix the existing files. If you can configure eslint then probably its fix option would do it, but I'd guess it will make a lot of mistakes.

From what I've read before, the best way to deal with code formatting is to use either standardjs or prettier and recently Biome is another option. However they can't be configured much and will format code in a way that some CC developers find unacceptable, so I'm not sure there is a viable option for automatically enforcing and fixing indentation.
